Source/Parallelization/WarpXComm_K.H

```diff
@@ -237,20 +231,18 @@ void warpx_interp (int j, int k, int l,
     kc = amrex::coarsen(kn, rk);
     lc = amrex::coarsen(ln, rl);
 
-    wj = 1.0_rt;
-    wk = 1.0_rt;
-    wl = 1.0_rt;
     for         (int jj = 0; jj < nj; jj++) {
         for     (int kk = 0; kk < nk; kk++) {
             for (int ll = 0; ll < nl; ll++) {
-                wj = (rj - amrex::Math::abs(jn - (jc + jj) * rj)) / static_cast<amrex::Real>(rj);
+                auto c = arr_coarse_zeropad(jc+jj,kc+kk,lc+ll);
+                c *= (rj - amrex::Math::abs(jn - (jc + jj) * rj)) / static_cast<amrex::Real>(rj);
 #if (AMREX_SPACEDIM >= 2)
-                wk = (rk - amrex::Math::abs(kn - (kc + kk) * rk)) / static_cast<amrex::Real>(rk);
+                c *= (rk - amrex::Math::abs(kn - (kc + kk) * rk)) / static_cast<amrex::Real>(rk);
```

Comment:
The test that triggered this is `clang-analyzer-deadcode.DeadStores`.  This test flags as issues scenarios like:
```
int a = 1;
# ifdef FLAG
    a = 2;
#endif

do_stuff(a);
```
if  `FLAG` is defined the value stored initially in `a` is never used.

Reply:
I think that in general it is a good test, although here it forces us to change a perfectly valid code. I do prefer the new version though, because  I sometimes find code like
```
int a = 1;
# ifdef FLAG
    a = 2;
#endif
```
a bit confusing.

Reply:
Of course there is no difference in terms of performances.

Reply:
Agreed, new looks better.